raiden/network/transport/matrix.py

```diff
@@ -258,31 +276,31 @@ def on_success(greenlet):
         self.greenlets.append(greenlet)
         return greenlet
 
+    @_lockedmethod(_factorygetter('__health_lock', Semaphore))
```

Comment:
TBH, i still think this is a bit complicated. This would have been replaced by:
1. `__init__` that declares `_health_lock`
2. a `with self._health_lock` block in the `start_health_check` method

Reply:
The whole point of these decorators is to avoid exactly that. I _could_ do it like that, or I _could_ do it like I did, making it explicit **and** declarative when a whole method is serialized. I know we could do it differently, but I don't agree that it's complicated, the helper decorators and functions are local and well documented.

Reply:
I agree this implementation is way too complicated, specially compared to what is used in trustlines.